### DISCUSSION TITLE
Widening and Narrowing

### DISCUSSION THREAD

**Discussion by HLe4s-kaist** [prosyslab-classroom/cs524-program-analysis#432] [Art competition]
![Widening and Narrowing](https://github.com/prosyslab-classroom/cs524-program-analysis/assets/161300429/3863c385-61e7-4a8b-94a4-5f7c4381595f)

(English follows)
## Name
강찬우

## Tool You Used
[Adobe Fire Fly](https://firefly.adobe.com/upload/inpaint), [bing image creator](https://www.bing.com/images/create)

## Short Description
매우 높고 속이 비쳐 보이는 빌딩이 하나 있고. 그 빌딩 중간에 커다랗고 예쁜 다이아몬드가 있다. 두 도둑이 빌딩의 양쪽 벽을 타며 이를 훔치려 하고 있다. 하나는 천장에서 내려오고 하나는 아래에서 올라간다. 

## Long Description (up to 3 sentences)
넓히기(Widening)와 좁히기(Narrowing)는 결국, 고정점(Fixed Point)을 찾기 위한 하나의 여정이라고 생각합니다. 이를 나타내기 위해서, 그 다이아몬드를 훔치려고 하는 도둑들이 위에서 내려오고 아래에서 올라가는 그림을 그려봤습니다. 프로그램 정적 분석가인 우리들도, 결국은 이 고정점을 손에 넣기 위해 요약 의미(Abstract Semantics)위를 타고 오르내리는 한 명의 도둑이지 않겠습니까.

[Eng]
## Name
ChanU Kang

## Tool You Used
[Adobe Fire Fly](https://firefly.adobe.com/upload/inpaint), [Bing Image Creator](https://www.bing.com/images/create)

## Short Description
There is a very tall and transparent building with a large, beautiful diamond in the middle. Two thieves are climbing the building's walls to steal it—one descending from the ceiling and one ascending from below.

## Long Description (up to 3 sentences)
I believe Widening and Narrowing are essentially journeys to find a Fixed Point. To illustrate this, I drew a picture of thieves descending from above and ascending from below to steal the diamond. As static program analyzers, aren't we like those thieves, ultimately climbing up and down Abstract Semantics to grasp the Fixed Point?

**Comment by KihongHeo** [comment]
국문으로 공식적인 글을 쓸 때에는, 한국어로 제대로 쓰고 전문 용어는 괄호안에 병기하는 습관을 기릅시다. 여러분들은 향후 사회 각계의 지도자가 될 텐데 지금부터 연습합시다. 예를 들어, 고정점(fixed point), 요약 의미(abstract semantics).
한번 고쳐 보세요. next time 부터 청조체는 not acceptable 입니다.

**Comment by HLe4s-kaist** [comment]
[Kor]
좋은 지적 해주셔서 감사합니다. 완전히 잊고 있었습니다. 청조체 수정했습니다 !
[Eng]
Thank you for pointing that out. I had completely forgotten about it. I've revised it in a formal tone!

**Comment by KihongHeo** [comment]
확대화 축소화 보다는 넓히기 좁히기가 더 와닿을것 같네요.
시야를 확 넓혀서 찾아보다가 너무 갔다 싶으면 다시 시야를 좁혀서 고정점 찾기.

**Comment by HLe4s-kaist** [comment]
[Kor]
너무 좋습니다 ! 훨신 와닿는거 같네요. 감사합니다 !
[Eng]
Thank you !
